### PR TITLE
Automatically update nuget packages

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Signed-off-by: Alan Jowett <alanjo@microsoft.com>

## Description

Automatically create PRs to update nuget packages to latest.

## Testing

CI/CD

## Documentation

No.

### CODE REVIEW

File: .github/workflows/nuget_update.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  schedule:
+    - cron: '0 21 * * 6'
+
+permissions:
+    contents: read
+
+jobs:
+  update:
+    permissions:
+      contents: write  # for Git to git push
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply .github/workflows/nuget_update.yaml:21]: This is required to be able to create PR with fixed dependencies.

File: .github/workflows/nuget_update.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      working-directory: ${{env.GITHUB_WORKSPACE}}
+      run: nuget restore -noninteractive -verbosity detailed ${{env.SOLUTION_FILE_PATH}}
+
+    # Update nuget packages used by the solution.
+    - name: Update NuGet packages
+      run: |
+        nuget update -noninteractive -verbosity detailed ${{env.SOLUTION_FILE_PATH}}
+
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v3
```
[the remark this answers is not in the document]
Comment (dthaler) [reply .github/workflows/nuget_update.yaml:59]: can this pin by hash instead of using v3?

**Review by dthaler — COMMENTED** [review]
Ok other than needing to pin by hash instead of version

**Comment by Alan-Jowett** [comment]
> Ok other than needing to pin by hash instead of version

Should pin by hash now.

**Review by dv-msft — APPROVED** [review]

**Pull request merged**